src/Collection/Collection.router.ts

```diff
@@ -376,42 +390,28 @@ export class CollectionRouter extends Router {
   deleteCollection = async (req: AuthRequest) => {
     const id = server.extractFromReq(req, 'id')
 
-    if (await this.isCollectionPublished(id)) {
+    const collection = (await Collection.findOne(id)) as CollectionAttributes // existance checked on middleware
+
+    if (await this.service.isPublished(collection.contract_address)) {
       throw new HTTPError(
         "The collection is published. It can't be deleted",
         { id },
         STATUS_CODES.unauthorized
       )
     }
 
+    if (this.service.isLockExpired(collection.lock)) {
+      throw new HTTPError(
+        "The collection is locked. It can't be deleted",
+        { id },
+        STATUS_CODES.unauthorized
```

Comment:
What do you think about returning a 423 status code (Locked)? Unauthorized refers to the authentication of the user.

Reply:
Much better 👍